src/kili/services/label_data_parsing/json_response.py

```diff
@@ -99,14 +198,55 @@ def __iter__(self) -> Iterator[str]:
         """Returns an iterator over the job names."""
         return iter(self._json_data)
 
-    def items(self) -> Iterator[Tuple[str, "job_response_module.JobPayload"]]:
-        """Returns an iterator over the job names and the corresponding JobPayload objects."""
-        return iter(self._json_data.items())
+    def items(self) -> Iterator[Tuple[str, "JobPayload"]]:
+        """Returns an iterator over the job names and the corresponding objects."""
+        return cast(Iterator[Tuple[str, "JobPayload"]], iter(self._json_data.items()))
 
     def keys(self) -> Iterator[str]:
         """Returns an iterator over the job names."""
         return iter(self._json_data.keys())
 
-    def values(self) -> Iterator["job_response_module.JobPayload"]:
-        """Returns an iterator over the JobPayload objects."""
-        return iter(self._json_data.values())
+    def values(self) -> Iterator["JobPayload"]:
+        """Returns an iterator over the objects."""
+        return cast(Iterator["JobPayload"], iter(self._json_data.values()))
+
+    def __getitem__(self, job_name: str) -> "JobPayload":
+        """Returns the object corresponding to the job name.
+
+        For a video job, the returned object is a FramesList.
+
+        Otherwise, the returned object is a JobPayload.
+        """
+        if job_name not in self._json_data:
+            raise JobNotExistingError(job_name)
+        return cast("JobPayload", self._json_data[job_name])
+
+
+class FramesList(List["job_response_module.JobPayload"]):
+    """List class that allows to access the JobPayload object corresponding to a frame number."""
+
+    def __getitem__(self, key: int) -> "job_response_module.JobPayload":
+        """Returns the JobPayload object corresponding to the frame number."""
+        if not 0 <= key <= len(self) - 1:
+            raise FrameIndexError(frame_index=key, nb_frames=len(self))
+        return super().__getitem__(key)
+
+    @property
+    def frames(self) -> "FramesList":
+        """Returns the list of frames."""
+        return self
+
+
+class JobPayload(FramesList, job_response_module.JobPayload):
```

Comment:
are you able to keep good type hints with this ?
It will always propose you the attibutes frames even if it is not a video project isn't it ?
(I don't see how to do it another way)

Reply:
we have all the possible fields (annotations, etc.), and this can only be decided at runtime, lgtm

Reply:
type hints work yes, otherwise pyright CI would fail

Well during run time, the program will crash if you try to access `label.jobs["job_name"].frames` on a non-video label yes

But since at static time, we do not know the label input type, we still have to expose the attribute `.frames`, for autocompletion, hover peeks, etc

and yes that's the only way I found to do it